### PR TITLE
[FEATURE] New component for search system

### PR DESCRIPTION
**Pull request by cregourd**
Integration of a new component for search system. This component is made by using React, Vite JS is used to bundle all React component into bundle.

- [x] Still possible to use plain HTML with the old system (no javascript enabled)
- [x] Use ViteJs for bundling
- [x] Allow multiple scoping levels for search

> ⚠️ This component is currently not functional because it requires a change to the  `search/suggest` API endpoint. All suggested packages must have a slug to redirect to the pre-filled search page.

### CODE REVIEW

**Comment by linawolf** [comment]
When you make changes to the templates please run 

```
make integration-baseline
```
to make the tests pass

File: packages/typo3-docs-theme/resources/js/components/GlobalSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    const [isModalOpen, setIsModalOpen] = useState(false);
+
+    const handleInputClick = () => {
+        setIsModalOpen(true);
+    };
+
+    useEffect(() => {
+        const globalSearchInput = document.getElementById('globalsearchinput');
+        globalSearchInput.addEventListener('click', handleInputClick);
+    }, []);
```
Comment (tlenclos) [inline packages/typo3-docs-theme/resources/js/components/GlobalSearch.jsx:14]:
```suggestion
useEffect(() => {
    const globalSearchInput = document.getElementById('globalsearchinput');

    if (globalSearchInput) {
        globalSearchInput.addEventListener('click', handleInputClick);
        return () => {
            globalSearchInput.removeEventListener('click', handleInputClick);
        };
    }
}, []);
```

**Review by tlenclos — CHANGES_REQUESTED** [review]

File: packages/typo3-docs-theme/resources/public/fonts/fa-brands-400.ttf
Comment (damienalexandre) [inline packages/typo3-docs-theme/resources/public/fonts/fa-brands-400.ttf:1]: As you can see, Font Awesome was not up to date with the version in packages.json.

```diff
- * Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com
+ * Font Awesome Free 6.7.0 by @fontawesome - https://fontawesome.com
```

So this PR contains the font update + CSS.

Should we split this in two PR or is it ok? @linawolf

File: packages/typo3-docs-theme/resources/js/components/SearchModal.jsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                url.searchParams.append('scope', encodeURIComponent(`/${scope.slug}/`));
+            } else if (scope.type === 'option') {
+                url.searchParams.append(`filters[optionaggs][${scope.title}]`, true);
+            } else {
+                url.searchParams.append(`filters[${scope.type}][${scope.title}]`, true);
+            }
+        });
+        url.searchParams.append('q', query);
+        return url.href;
+    }, []);
```
Comment (damienalexandre) [inline packages/typo3-docs-theme/resources/js/components/SearchModal.jsx:34]: We miss something here.

In suggestions, we need the manual_slug, to build proper contextualized urls.

File: packages/typo3-docs-theme/resources/js/components/SuggestRow.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    if (scopes?.length > 0) {
+        return (
+            <>
+                {scopes.map(({ title: scopeName, type }) => (
+                    <div key={`${type}-${scopeName}`} className='suggest-row__scope'>
+                        <p className="suggest-row__scope-type">{type && `${type}:`}</p>
+                        {scopeName && <p className="suggest-row__scope-name">{scopeName}</p>}
+                    </div>
+                ))}
+                <p className="suggest-row__title">{title}</p>
```
Comment (damienalexandre) [inline packages/typo3-docs-theme/resources/js/components/SuggestRow.jsx:21]: We may need to add an ellipsis here for long titles on mobile.

File: packages/typo3-docs-theme/resources/js/hooks/useSearchSuggestions.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            
+            if (!response.ok) throw new Error('Network response error');
+            
+            const data = await response.json();
+            
+            // Process file suggestions
+            const files = data?.results?.map(result => ({
+                title: result.snippet_title,
+                packageName: result.manual_package,
+                href: `http://localhost:300°/${result.manual_slug}/${result.relative_url}#${result.fragment}`,
```
Comment (damienalexandre) [inline packages/typo3-docs-theme/resources/js/hooks/useSearchSuggestions.js:48]: This needs to be replaced.

File: packages/typo3-docs-theme/resources/public/fonts/fa-brands-400.ttf
Comment (linawolf) [reply packages/typo3-docs-theme/resources/public/fonts/fa-brands-400.ttf:1]: This is fine I think

File: packages/typo3-docs-theme/resources/js/components/SearchModal.jsx (hunk 2 above)
Comment (damienalexandre) [reply packages/typo3-docs-theme/resources/js/components/SearchModal.jsx]: Done :white_check_mark:

**Review by linawolf — APPROVED** [review]

**Pull request merged**